src/main/groovy/wooga/gradle/node/NodeReleasePlugin.groovy

```diff
@@ -123,19 +126,27 @@ class NodeReleasePlugin implements Plugin<Project> {
         })
     }
 
-    private static void configureNpmCredentialsTasks(Project project, NodeReleasePluginExtension extension) {
-        project.tasks.withType(NpmCredentialsTask, new Action<NpmCredentialsTask>() {
-
-            @Override
-            void execute(NpmCredentialsTask npmCredentialsTask) {
-                npmCredentialsTask.group = TASK_GROUP
-                npmCredentialsTask.description = "create ${NPMRC} file"
-                npmCredentialsTask.npmUser.set(extension.npmUser)
-                npmCredentialsTask.npmPass.set(extension.npmPass)
-                npmCredentialsTask.npmAuthUrl.set(extension.npmAuthUrl)
-                npmCredentialsTask.npmrcFile.set(extension.npmrcFile)
+    private void configureNpmCredentialsTasks(Project project, NodeReleasePluginExtension extension) {
+        project.tasks.withType(NpmCredentialsTask).configureEach { task ->
+            task.group = TASK_GROUP
+            task.description = "create ${NPMRC} file"
+            task.npmUser.set(extension.npmUser)
+            task.npmPass.set(extension.npmPass)
+            task.npmAuthUrl.set(extension.npmAuthUrl)
+            task.npmrcFile.set(extension.npmrcFile)
+            task.onlyIf {
+                def noHomeNpmRC = project.provider { !new File(System.getProperty("user.home"), ".npmrc").file }
+                        .map(warnFalse{".npmrc file present on user home dir, skipping"})
+                        .orElse(true)
```

Comment:
I wouldn't check if the home has an npmrc file. That was never the intention since it could also miss the credentials for the project. Just run the task only if credentials are provided. If they are missing then it is up to the user who he sees fit to provide it. Either by setting them or adding an nprmrc file manully to either the home or project folder.

Reply:
Yeah, makes sense for given credentials not be overriden by some external factor. Done.